guardian/mobile-apps-api-models
# Add list object to collection, rename protofile

## What does this change?
 
This PR adds a list object to <strike>collection.proto</strike> blueprint.proto so that we can support lists/tags in blueprint. It also renamed the protofile from collection.proto to blueprint.proto, to better semantically represent our models now that we include a list and collection object (which are on the same level in terms of heirachy). I've also added some re-ordered the proto file and added comments to hopefully make it easier to read. 

## How to test

I've released a snapshot, updated my locally running MAPI and tested on the iOS simulator. All works as expected. The only difference is our imports will now look like `import com.gu.mobile.mapi.models.v0.blueprint` as opposed to `import com.gu.mobile.mapi.models.v0.collection`
